tests/test_calc.py

```diff
@@ -348,9 +348,7 @@ def test_other_functions(self):
         self.assert_function_values('factorial', fact_inputs, fact_values)
 
         self.assertRaises(ValueError, calc.evaluator, {}, {}, "fact(-1)")
-        self.assertRaises(ValueError, calc.evaluator, {}, {}, "fact(0.5)")
```

Comment:
Why were these assertions removed?

Reply:
@feanil  python 3.12 will not allow math.factorial to calculate float values and will raise an error [ref](https://docs.python.org/3.12/library/math.html#math.factorial) 

So earlier if we sent integers like `1.0` it'll work but if we sent float `1.5` it used to give value error 
but now if we send integers like `1.0` it'll give error, and because of that I wrote extra code to check if it's factorial we'll convert all incoming numbers to integers, because `calc.py`  has `evaluator` function which converts all numbers to floats. And if we send it `0.5` it'll just take it as `0` and the test won't work..

Reply:
@feanil, yes, 

I'll try to explain my changes. 

Python 3.8 

The value we were sending from here [test_calc.py](https://github.com/openedx/openedx-calc/blob/master/tests/test_calc.py#L345-L348)

Were being sent as `0.0` or `1.0` `5.0` etc and that was allowed to be calculated in python 3.8 factorial.

Python 3.12

While the values being sent are still the same i.e. `0.0` or `1.0`, it's no longer allowed in python 3.12 to pass them like that in float, we need to pass them as `0` or `1` like integers, as a result I made this change, to convert these values to integer if the function being called is integer [change ref](https://github.com/openedx/openedx-calc/pull/99/files#diff-ac76f42eb0f507408ecbd6c2242a4e1c0b97b7e12d0db1aabd392f457a625dc7R269)

What happens now is `0.5` will be parsed as `0`, because I am now converting things passed for factorials as integers. Thus I removed it.

As for the error output if we don't make changes, it is something like this. 

```
======================================================================
ERROR: test_other_functions (tests.test_calc.EvaluatorTest.test_other_functions)
Test the non-trig functions provided in calc.py
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/chintan/openedx-repos/openedx-calc_old/tests/test_calc.py", line 347, in test_other_functions
    self.assert_function_values('fact', fact_inputs, fact_values)
  File "/home/chintan/openedx-repos/openedx-calc_old/tests/test_calc.py", line 168, in assert_function_values
    result = calc.evaluator({}, {}, input_str)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/chintan/openedx-repos/openedx-calc_old/calc/calc.py", line 281, in evaluator
    return math_interpreter.reduce_tree(evaluate_actions)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/chintan/openedx-repos/openedx-calc_old/calc/calc.py", line 454, in reduce_tree
    return handle_node(self.tree)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/chintan/openedx-repos/openedx-calc_old/calc/calc.py", line 450, in handle_node
    handled_kids = [handle_node(k) for k in node]
                    ^^^^^^^^^^^^^^
  File "/home/chintan/openedx-repos/openedx-calc_old/calc/calc.py", line 450, in handle_node
    handled_kids = [handle_node(k) for k in node]
                    ^^^^^^^^^^^^^^
  File "/home/chintan/openedx-repos/openedx-calc_old/calc/calc.py", line 450, in handle_node
    handled_kids = [handle_node(k) for k in node]
                    ^^^^^^^^^^^^^^
  [Previous line repeated 2 more times]
  File "/home/chintan/openedx-repos/openedx-calc_old/calc/calc.py", line 451, in handle_node
    return action(handled_kids)
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/chintan/openedx-repos/openedx-calc_old/calc/calc.py", line 268, in eval_function
    return all_functions[casify(x[0])](x[1])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: 'float' object cannot be interpreted as an integer

```

Reply:
@ichintanjoshi I dug into it a little deeper, based on your stack trace, it looks like the error is happening with the call on line 347 and not on the line that you removed.  I don't think the fix you put in place is the right one because we don't want to coerce actual floats to integers as that will fail to raise errors when it correctly should.

I think the real problem is with the [eval_number](https://github.com/openedx/openedx-calc/blob/master/calc/calc.py#L123-L130) function which coerces all numbers to floats.  I think that function needs to be smarter and if there are no decimals in the number then it needs to return integers instead of numbers.

Reply:
Yes, that was the issue, Okay I'll take a look into it if I can fix it.